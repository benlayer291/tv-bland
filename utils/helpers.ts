export const removeProtocolFromUrl = (url: string): string =>
  url.replace(/^https?:\/\//i, '//')

export const removeHtmlTagsFromString = (htmlString: string): string =>
  htmlString.replace(/(<([^>]+)>)/gi, '')

export const checkDeviceIsMobile = (): boolean => {
  return (
    typeof navigator !== 'undefined' &&
    /Android|Mobi/i.test(navigator.userAgent)
  )
}
