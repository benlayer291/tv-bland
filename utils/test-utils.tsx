import { render, RenderOptions } from '@testing-library/react'

type Props = {
  children: React.ReactElement
}

const Providers: React.FunctionComponent<Props> = ({ children }: Props) => {
  return <>{children}</>
}

const customRender = (
  ui: React.ReactElement,
  options: Omit<RenderOptions, 'queries'> = {}
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
