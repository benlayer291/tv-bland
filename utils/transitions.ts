type Transtion = {
  duration: number
  ease: number[]
}

export const transition: Transtion = {
  duration: 5,
  ease: [0.43, 0.13, 0.23, 0.96],
}
