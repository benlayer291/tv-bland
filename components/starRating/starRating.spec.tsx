import React from 'react'
import { render, RenderResult } from '../../utils/test-utils'

import StarRating from './starRating'

const propsDefault = {
  rating: 8,
}

describe('<StarRating />', () => {
  let el: RenderResult

  it('should render', () => {
    el = render(<StarRating {...propsDefault} />)

    expect(el.container).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    el = render(<StarRating {...propsDefault} />)

    expect(el.container).toMatchSnapshot()
  })

  describe('given "withText" is true', () => {
    it('should render with text', () => {
      el = render(<StarRating {...propsDefault} withText={true} />)

      expect(el.queryByTestId('rating-text')).toBeInTheDocument()
    })
  })

  describe('given "islarge" is true', () => {
    it('should render with "large" styles', () => {
      el = render(<StarRating {...propsDefault} isLarge={true} />)

      expect(el.queryByTestId('rating')).toHaveClass('StarRating--is-large')
    })
  })
})
