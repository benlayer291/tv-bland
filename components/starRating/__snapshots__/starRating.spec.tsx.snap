// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarRating /> should match snapshot 1`] = `
<div>
  <div
    class="StarRating"
    data-testid="rating"
  >
    <div
      class="StarRating__icons"
    >
      <svg
        class="Star StarRating__icon Star--is-active"
        data-testid="star"
        viewBox="0 0 14 12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M6.667 9.95L2.547 12l.786-4.342L0 4.584l4.607-.634L6.667 0l2.06 3.95 4.606.634L10 7.658 10.787 12l-4.12-2.05z"
          fill-rule="evenodd"
        />
      </svg>
      <svg
        class="Star StarRating__icon Star--is-active"
        data-testid="star"
        viewBox="0 0 14 12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M6.667 9.95L2.547 12l.786-4.342L0 4.584l4.607-.634L6.667 0l2.06 3.95 4.606.634L10 7.658 10.787 12l-4.12-2.05z"
          fill-rule="evenodd"
        />
      </svg>
      <svg
        class="Star StarRating__icon Star--is-active"
        data-testid="star"
        viewBox="0 0 14 12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M6.667 9.95L2.547 12l.786-4.342L0 4.584l4.607-.634L6.667 0l2.06 3.95 4.606.634L10 7.658 10.787 12l-4.12-2.05z"
          fill-rule="evenodd"
        />
      </svg>
      <svg
        class="Star StarRating__icon Star--is-active"
        data-testid="star"
        viewBox="0 0 14 12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M6.667 9.95L2.547 12l.786-4.342L0 4.584l4.607-.634L6.667 0l2.06 3.95 4.606.634L10 7.658 10.787 12l-4.12-2.05z"
          fill-rule="evenodd"
        />
      </svg>
      <svg
        class="Star StarRating__icon"
        data-testid="star"
        viewBox="0 0 14 12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M6.667 9.95L2.547 12l.786-4.342L0 4.584l4.607-.634L6.667 0l2.06 3.95 4.606.634L10 7.658 10.787 12l-4.12-2.05z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;
